apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -144,28 +145,36 @@ message_stats() ->
 %%--------------------------------------------------------------------------------
 
 %% @doc Called when a client connects. This function looks up a
-%% session or creates a new one if previous one couldn't be found.
+%% session or returns `false` if previous one couldn't be found.
 %%
 %% This function also spawns replay agents for each iterator.
 %%
 %% Note: session API doesn't handle session takeovers, it's the job of
 %% the broker.
--spec session_open(session_id(), _Props :: map()) ->
-    {_New :: boolean(), session(), iterators()}.
-session_open(SessionId, Props) ->
+-spec session_open(session_id()) ->
+    {ok, session(), iterators()} | false.
+session_open(SessionId) ->
     transaction(fun() ->
         case mnesia:read(?SESSION_TAB, SessionId, write) of
             [Record = #session{}] ->
                 Session = export_record(Record),
                 IteratorRefs = session_read_iterators(SessionId),
                 Iterators = export_iterators(IteratorRefs),
-                {false, Session, Iterators};
+                {ok, Session, Iterators};
             [] ->
-                Session = export_record(session_create(SessionId, Props)),
-                {true, Session, #{}}
+                false
         end
     end).
 
+-spec session_ensure(session_id(), _Props :: map()) ->
+    {ok, session(), iterators()}.
+session_ensure(SessionId, Props) ->
```

Comment:
Nit: the function name gives me the impression that it would have the previous `session_open` semantics: if a matching session already exists, it's returned; otherwise, a fresh one is returned; rather than always returning a fresh one.  :thinking: 

Perhaps:

```suggestion
session_ensure_new(SessionId, Props) ->
```

Reply:
✅ a2ddd9d5f.